src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionTransfer.java

```diff
@@ -48,31 +47,68 @@ public TransferResult<CFValue, CFStore> visitMethodInvocation(
               boundClassTypeMirror.getAnnotation(), diATF.classValValueElement, String.class);
 
       accumulate(node, result, classNames.toArray(new String[1]));
+
     } else if (diATF.isAnnotatedWithMethod(node.getTree())) {
+
       MethodInvocationNode methodInvocationNode = node;
-      MethodInvocationNode callToBindNode =
-          (MethodInvocationNode) methodInvocationNode.getTarget().getReceiver();
-      MethodInvocationNode namesClassNode =
-          (MethodInvocationNode) methodInvocationNode.getArgument(0);
-      FieldAccessNode boundClass = (FieldAccessNode) callToBindNode.getArgument(0);
-      StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
-
-      accumulateBindAnnotatedWith(
-          node, result, boundClass.getReceiver().getType().toString(), givenName.getValue());
+
+      CFValue value =
+          diATF
+              .getStoreBefore(node)
+              .getValue(JavaExpression.fromNode(methodInvocationNode.getTarget().getReceiver()));
+
+      if (value != null && !value.getAnnotations().isEmpty()) {
+        AnnotationMirror anno = value.getAnnotations().first();
+        List<String> classNames =
+            AnnotationUtils.getElementValueArray(anno, diATF.bindValValueElement, String.class);
```

Comment:
This line makes the strong assumption that the first annotation will definitely be `@Bind`. I think you should test that before calling `getElementValueArray`: it's probably true almost all the time (`@Bind({})` is the default), but there's no guarantee that e.g. the user hasn't provided a weird stub file or something like that. You should always test what kind of annotation you have before calling `getElementValue` or `getElementValueArray`.